### PR TITLE
Change MySQL ENUM columns to have base_type=type/TextLike

### PR DESCRIPTION
**Pull request by calherries**
Solves https://github.com/metabase/metabase/issues/37638

This should be backported with https://github.com/metabase/metabase/pull/43483, because otherwise fingerprinting and scanning won't happen anymore with ENUM columns.

MySQL ENUM columns have always had `base_type=type/*`, even though they are _mostly_ indistinguishable from other TEXT columns. This PR changes that so ENUM columns have `base_type=type/TextLike`.

After a discussion with Braden and Dan [on slack](https://metaboat.slack.com/archives/CKZEMT1MJ/p1717443858196669) we decided that `type/TextLike` was more appropriate for these columns like this. Unlike `type/Text`, `type/TextLike` disallows the application of text operations like "contains" or "starts-with" filters, or using functions like "concat". See [this code](https://github.com/metabase/metabase/blob/2a21cf2ea2e1708ce4f4b4fdea77ea96189a852c/src/metabase/lib/schema/expression/string.cljc#L1) for how this works in the QP.

For more information on ENUM columns, see the [MySQL docs](https://dev.mysql.com/doc/refman/8.4/en/enum.html). tl;dr is they are mostly similar to TEXT columns but differ in a few ways. For example, an `ORDER BY` clause will order the column by the integer values that back the string values, not the string values themselves.

To test this manually, create a table in a MySQL DB with the following SQL:
```
-- Create the table
CREATE TABLE colors (
    id INT AUTO_INCREMENT PRIMARY KEY,
    color ENUM('red', 'green', '1')
);

-- Insert values into the table
INSERT INTO colors (color) VALUES ('red');
INSERT INTO colors (color) VALUES ('green');
INSERT INTO colors (color) VALUES ('1');
INSERT INTO colors (color) VALUES (NULL);
```

### CODE REVIEW

**Comment by qnkhuat** [comment]
I think this needs a test checking that enum columns are synced correctly

**Comment by calherries** [comment]
@qnkhuat I don't grok what exactly you mean by "synced correctly". What property of a synced enum column would you like to test?

**Review by bshepherdson — APPROVED** [review]

**Comment by calherries** [comment]
@qnkhuat I've added a sync and actions test for ENUM columns, can you take another look?

File: test/metabase/driver/mysql_test.clj [hunk 1]
```diff
@@ -361,6 +362,47 @@
...
+  (mt/test-driver :mysql
+    (testing "ENUM columns are synced with the correct base and semantic types"
+      (mt/with-empty-db
+        (create-enums-table! (mt/db))
+        (sync/sync-database! (mt/db))
+        (is (=? {:base_type     :type/MySQLEnum
+                 :semantic_type :type/Category}
+                (t2/select-one :model/Field :name "bird_type")))))))
+
+(deftest enums-actions-test
```
Comment (calherries) [inline test/metabase/driver/mysql_test.clj:381]: This test is for https://github.com/metabase/metabase/issues/37638

**Review by qnkhuat — APPROVED** [review]
nice. Thanks for the test!

**Comment by calherries** [comment]
@metabase-bot backport release-x.49.x

**Pull request merged**